src/Middleware/HttpLogging/src/FileLoggerProcessor.cs

```diff
@@ -0,0 +1,240 @@
+                    return;
+                }
+                fileInfo.Refresh();
+                // Roll to new file if _maxFileSize is reached
+                // _maxFileSize could be less than the length of the file header - in that case we still write the first log message before rolling
+                if (fileInfo.Exists && fileInfo.Length > _maxFileSize)
+                {
+                    _fileNumber++;
+                    fullName = GetFullName(today);
+                    fileInfo = new FileInfo(fullName);
+                    streamWriter.Dispose();
+                    if (!TryCreateDirectory())
+                    {
+                        // return early if we fail to create the directory
+                        return;
+                    }
+                    streamWriter = GetStreamWriter(fullName);
+                }
+                if (!fileInfo.Exists || fileInfo.Length == 0)
+                {
+                    await OnFirstWrite(streamWriter);
+                }
+
+                await WriteMessageAsync(message, streamWriter);
+            }
+            streamWriter.Dispose();
+
+            RollFiles();
+        }
+
+        internal bool TryCreateDirectory()
+        {
+            if (!Directory.Exists(_path))
+            {
+                try
+                {
+                    Directory.CreateDirectory(_path);
+                    return true;
+                }
+                catch (Exception ex)
+                {
+                    Log.CreateDirectoryFailed(_logger, _path, ex);
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        // Virtual for testing
+        internal virtual async Task WriteMessageAsync(string message, StreamWriter streamWriter)
+        {
+            await streamWriter.WriteLineAsync(message);
+            await streamWriter.FlushAsync();
+        }
+
+        // Virtual for testing
+        internal virtual StreamWriter GetStreamWriter(string fileName)
+        {
+            return File.AppendText(fileName);
+        }
+
+        private void RollFiles()
+        {
+            if (_maxRetainedFiles > 0)
+            {
+                var files = new DirectoryInfo(_path)
+                    .GetFiles(_fileName + "*")
+                    .OrderByDescending(f => f.Name)
+                    .Skip(_maxRetainedFiles.Value);
+
+                foreach (var item in files)
+                {
+                    item.Delete();
+                }
+            }
+        }
+
+        public async ValueTask DisposeAsync()
+        {
+            _cancellationTokenSource.Cancel();
+            _messageQueue.CompleteAdding();
+            _outputTask.Wait();
+            await ValueTask.CompletedTask;
+        }
+
+        private string GetFullName(DateTime date)
+        {
+            return Path.Combine(_path, $"{_fileName}{date.Year:0000}{date.Month:00}{date.Day:00}.{_fileNumber}.txt");
```

Comment:
Is this going to output localized numbers if you don't use ToString(InvariantCulture)?

Reply:
Yeah, `Year`, `Month`, and `Day` are ints so would be localized for Cultures using different numerals - are you saying the filenames should always use InvariantCulture numerals?

Reply:
Prefer InvariantCulture unless you have a specific requirement otherwise.

Reply:
https://github.com/dotnet/aspnetcore/pull/33251/commits/9c1d1b50ca6b1a2eef862c29d0f32b4528154f43